packages/rpc-provider/src/ws/Provider.ts

```diff
@@ -138,7 +141,9 @@ export default class WsProvider implements WSProviderInterface {
 
       const WS = await getWSClass();
 
-      this.#websocket = new WS(this.#endpoints[this.#endpointIndex]);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore - WS may be an instance of w3cwebsocket, which supports headers
+      this.#websocket = new WS(this.#endpoints[this.#endpointIndex], null, null, this.#headers);
```

Comment:
```suggestion
      this.#websocket = new WS(this.#endpoints[this.#endpointIndex], undefined, undefined, this.#headers);
```

type definition is `constructor(
        url: string,
        protocols?: string | string[],
        origin?: string,
        headers?: http.OutgoingHttpHeaders,
        requestOptions?: object,
        IClientConfig?: IClientConfig,
    );`

Reply:
Would also suggest, looking at the above -

```js
import { isChildClass } from '@polkadot/util`;

...
this.#websocket = typeof WebSocket !== 'undefined' && isChildClass(WebSocket, WS)
  ? new WS(this.#endpoints[this.#endpointIndex])
  // // @ts-ignore
  : new WS(this.#endpoints[this.#endpointIndex], undefined, undefined, this.#headers);
```

this should (hopefully, maybe) fix the test failures as well?

Reply:
Updated!